CSharp/BatchExplorer/Views/MainView.xaml

```diff
@@ -687,12 +658,20 @@
                                         <TextBlock HorizontalAlignment="Center" Text="Delete" FontSize="12" Foreground="Black" />
                                     </StackPanel>
                                 </Button.Content>
-                            </Button>                            
-                        </StackPanel>
+                            </Button>
+                            <TextBox x:Name="SearchJobsTxtBox" 

```

Comment:
Ah.. I missed this: "e.Key == Key.Enter". I thought you were calling the filter on every key stroke.

So, now that I've read this correctly, two things. 
1. yes, you do still need to call the code-behind. I would have though you'd tie into the PropertyChanged event trigger rather than the KeyUP, but in the end it's pretty much the same (you can see more here: http://stackoverflow.com/questions/563195/bind-textbox-on-enter-key-press). I'll argue that on line 463 of the code-behind you can move the "this.viewModel.JobsSearchFilter = this.SearchJobsTxtBox.Text" inside the if block if you only plan on handling this on enter.
2. another approach you can take (albeit a bit more complicated) is to filter-as-you-type. So do a 200 millisecond wait or something, and if there are not more key strokes then apply the filter. You can find more info on that here:
https://msdn.microsoft.com/en-us/library/ms742880(v=vs.110).aspx and http://stackoverflow.com/questions/8001450/c-sharp-wait-for-user-to-finish-typing-in-a-text-box.

Note, I'm not asking you to implement #2, just saying it's another option which is a bit cleaner IMO since you get rid of the explicit 'ENTER'.